packages/amazonq/test/unit/codewhisperer/models/model.test.ts

```diff
@@ -70,4 +75,113 @@ describe('model', function () {
             assert.deepStrictEqual(hiddenSeverities, ['High', 'Low'])
         })
     })
+
+    describe('CodeIssueGroupingStrategyState', function () {
+        let sandbox: sinon.SinonSandbox
+        let state: CodeIssueGroupingStrategyState
+
+        beforeEach(function () {
+            sandbox = sinon.createSandbox()
+            // Reset the singleton instance before each test
+            // @ts-ignore - accessing private static for testing
+            CodeIssueGroupingStrategyState['#instance'] = undefined
+            state = CodeIssueGroupingStrategyState.instance
+        })
+
+        afterEach(function () {
+            sandbox.restore()
+        })
+
+        describe('instance', function () {
+            it('should return the same instance when called multiple times', function () {
+                const instance1 = CodeIssueGroupingStrategyState.instance
+                const instance2 = CodeIssueGroupingStrategyState.instance
+                assert.strictEqual(instance1, instance2)
+            })
+        })
+
+        describe('getState', function () {
+            it('should return fallback when no state is stored', function () {
+                const tryGetStub = sandbox.stub(globals.globalState, 'tryGet').returns(undefined)
+                const result = state.getState()
+
+                sinon.assert.calledWith(tryGetStub, 'aws.amazonq.codescan.groupingStrategy', String)
```

Comment:
these kinds of tests are very fragile. can you assert the actual state instead of using stubs/mocks?